SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestCharacterMovement/SpatialTestCharacterMovement.cpp

```diff
@@ -36,38 +37,24 @@ ASpatialTestCharacterMovement::ASpatialTestCharacterMovement()
 	Description = TEXT("Test Character Movement");
 }
 
-void ASpatialTestCharacterMovement::OnOverlapBegin(AActor* OverlappedActor, AActor* OtherActor)
+bool ASpatialTestCharacterMovement::HasCharacterReachedDestination(ATestMovementCharacter* PlayerCharacter,
+																   const FPlane& DestinationPlane) const
 {
-	ATestMovementCharacter* OveralappingCharacter = Cast<ATestMovementCharacter>(OtherActor);
-
-	if (OveralappingCharacter)
-	{
-		bCharacterReachedDestination = true;
-	}
+	// Checks if the character has passed the plane
+	return DestinationPlane.PlaneDot(PlayerCharacter->GetActorLocation()) > 0;
 }
 
 void ASpatialTestCharacterMovement::PrepareTest()
 {
 	Super::PrepareTest();
 
-	// Universal setup step to create the TriggerBox and to set the helper variable
-	AddStep(TEXT("UniversalSetupStep"), FWorkerDefinition::AllWorkers, nullptr, [this]() {
-		bCharacterReachedDestination = false;
-
-		ATriggerBox* TriggerBox =
-			GetWorld()->SpawnActor<ATriggerBox>(FVector(232.0f, 0.0f, 40.0f), FRotator::ZeroRotator, FActorSpawnParameters());
+	FVector Origin = FVector(0.0f, 0.0f, 50.0f);
```

Comment:
Should we position the origin a bit further over the boundary? Maybe this is why I left a 20 unit overlap

Reply:
The origin is just used as the start position for the character, if we want it to move further we can add 20 to the destination?

Reply:
Is the 20cm overlap just a leftover from the bounding box? That was "tolerance" for low FPS. Testing a plane is akin to infinte tolerance.

Reply:
hmm, yeah - seems redundant